cyclonedx-lib/sign_src/TemurinSignSBOM.java

```diff
@@ -0,0 +1,191 @@
+import org.webpki.json.JSONParser;
+import org.webpki.util.PEMDecoder;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.StringReader;
+import java.io.IOException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.PublicKey;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.logging.Logger;
+import java.util.Base64;
+
+public final class TemurinSignSBOM {
+
+    private static boolean verbose = false;
+
+    // Create a logger for the class
+    private static final Logger LOGGER = Logger.getLogger(TemurinSignSBOM.class.getName());
+
+    private TemurinSignSBOM() {
+    }
+
+    /**
+     * Main entry.
+     * @param args Arguments for sbom operation.
+     */
+    public static void main(final String[] args) {
+        String cmd = null;
+        String privateKeyFile = null;
+        String publicKeyFile = null;
+        String fileName = null;
+
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equals("--jsonFile")) {
+                fileName = args[++i];
+            } else if (args[i].equals("--privateKeyFile")) {
+                privateKeyFile = args[++i];
+            } else if (args[i].equals("--publicKeyFile")) {
+                publicKeyFile = args[++i];
+            } else if (args[i].equals("--signSBOM")) {
+                cmd = "signSBOM";
+            } else if (args[i].equals("--verifySignature")) {
+                cmd = "verifySignature";
+            } else if (args[i].equals("--verbose")) {
+                verbose = true;
+            }
+        }
+
+
+        if (cmd.equals("signSBOM")) {
+            Bom bom = signSBOM(fileName, privateKeyFile);
+            writeJSONfile(bom, fileName);
+        } else if (cmd.equals("verifySignature")) {
+            boolean isValid = verifySignature(fileName, publicKeyFile);
+            System.out.println("Signature verification result: " + (isValid ? "Valid" : "Invalid"));
+        }
+    }
+
+    static Bom signSBOM(final String jsonFile, final String pemFile) {
+        try {
+            // Read the JSON file to be signed
+            Bom bom = readJSONfile(jsonFile);
+            String sbomDataToSign = generateBomJson(bom);
+
+            // Read the private key
+            File privateKeyFile = new File(pemFile);
+            byte[] privateKeyData = new byte[(int) privateKeyFile.length()];
+            FileInputStream privateKeyFileInputStream = new FileInputStream(privateKeyFile);
+            try {
+                privateKeyFileInputStream.read(privateKeyData);
+            } finally {
+                privateKeyFileInputStream.close();
+            }
+            String privateKeyString = new String(privateKeyData, "UTF-8");
+            KeyPair sampleKey = PEMDecoder.getKeyPair(privateKeyData);
```

Comment:
Replace the above code to use PEMDecoder utility and rename variable to "signingKey" rather than "sampleKey":
```
            // Read the private key
            KeyPair signingKey = PEMDecoder.getKeyPair(Files.readAllBytes(Paths.get(pemFile)));
```